src/test/kotlin/com/example/api/controller/BookingControllerIntegrationTest.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.api.controller
+import com.example.api.model.Customer
+import com.example.api.controller.request.BookingRequest
+import com.example.api.model.Booking
+import com.example.api.service.BookingService
+import org.junit.jupiter.api.Test
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.*
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+
+@ExtendWith(SpringExtension::class)
+@WebMvcTest(controllers = [BookingController::class])
+@ContextConfiguration
+class BookingControllerIntegrationTest {
+    @Autowired
+    private lateinit var mockMvc: MockMvc
+
+    @Autowired
+    private lateinit var objectMapper: ObjectMapper
+
+    @MockkBean
+    private lateinit var bookingService: BookingService
+
+    @Test
+    internal fun `Should create booking`(){
+        every { bookingService.addBooking(1, fakeBookingRequest) }
+
+        mockMvc.perform(
+            MockMvcRequestBuilders
+                .post("/customer/${fakeCustomer.customerId}/booking")
+                .content(objectMapper.writeValueAsBytes(fakeBookingRequest))
+                .contentType(MediaType.APPLICATION_JSON)
+        ).andExpect(MockMvcResultMatchers.status().isCreated)
```

Comment:
Your controller has response ok and here you've written isCreated. I don't think this test will pass. You'll have to use isOk here also.